tfhe/c_api_tests/test_high_level_integers.c

```diff
@@ -0,0 +1,201 @@
+    int ok;
+    FheUint8 *lhs = NULL;
+    FheUint8 *rhs = NULL;
+    FheUint8 *result = NULL;
+
+    uint8_t lhs_clear = 123;
+    uint8_t rhs_clear = 14;
+
+    ok = fhe_uint8_try_encrypt_with_public_key_u8(lhs_clear, public_key, &lhs);
+    GOTO_IF_TRUE(exit, ok != 0)
+
+    ok = fhe_uint8_try_encrypt_with_public_key_u8(rhs_clear, public_key, &rhs);
+    GOTO_IF_TRUE(exit, ok != 0)
+
+    ok = fhe_uint8_sub(lhs, rhs, &result);
+    GOTO_IF_TRUE(exit, ok != 0)
+
+    uint8_t clear;
+    ok = fhe_uint8_decrypt(result, client_key, &clear);
+    GOTO_IF_TRUE(exit, ok != 0)
+
+    assert(clear == (lhs_clear - rhs_clear));
+exit:
+    fhe_uint8_destroy(lhs);
+    fhe_uint8_destroy(rhs);
+    fhe_uint8_destroy(result);
+    return ok;
+}
+
+int uint8_serialization(const ClientKey *client_key) {
+    int ok;
+    FheUint8 *lhs = NULL;
+    FheUint8 *deserialized_lhs = NULL;
+    FheUint8 *result = NULL;
+    Buffer value_buffer = {.pointer = NULL, .length = 0};
+    Buffer cks_buffer = {.pointer = NULL, .length = 0};
+    BufferView deser_view = {.pointer = NULL, .length = 0};
+    ClientKey *deserialized_client_key = NULL;
+
+    uint8_t lhs_clear = 123;
+
+    ok = client_key_serialize(client_key, &cks_buffer);
+    GOTO_IF_TRUE(exit, ok != 0)
+
+    deser_view.pointer = value_buffer.pointer;
+    deser_view.length = value_buffer.length;
+    ok = client_key_deserialize(deser_view, &deserialized_client_key);
+    GOTO_IF_TRUE(exit, ok != 0)
+
+    ok = fhe_uint8_try_encrypt_with_client_key_u8(lhs_clear, deserialized_client_key, &lhs);
+    GOTO_IF_TRUE(exit, ok != 0)
+
+    ok = fhe_uint8_serialize(lhs, &value_buffer);
+    GOTO_IF_TRUE(exit, ok != 0)
+
+    deser_view.pointer = value_buffer.pointer;
+    deser_view.length = value_buffer.length;
+    ok = fhe_uint8_deserialize(deser_view, &deserialized_lhs);
+    GOTO_IF_TRUE(exit, ok != 0)
+
+    uint8_t clear;
+    ok = fhe_uint8_decrypt(deserialized_lhs, deserialized_client_key, &clear);
+    GOTO_IF_TRUE(exit, ok != 0)
+
+    assert(clear == lhs_clear);
+    
+ exit:
+    if (value_buffer.pointer != NULL) {
+        destroy_buffer(&value_buffer);
+    }
+    fhe_uint8_destroy(lhs);
+    fhe_uint8_destroy(deserialized_lhs);
+    fhe_uint8_destroy(result);
+  return ok;
+}
+
+int uint8_compressed(const ClientKey *client_key) {
+    int ok;
+    FheUint8 *lhs = NULL;
+    FheUint8 *result = NULL;
+    CompressedFheUint8 *clhs = NULL;
+
+    uint8_t lhs_clear = 123;
+
+    ok = compressed_fhe_uint8_try_encrypt_with_client_key_u8(lhs_clear, client_key, &clhs);
+    GOTO_IF_TRUE(exit, ok != 0)
+
+    ok = compressed_fhe_uint8_decompress(clhs, &lhs);
+    GOTO_IF_TRUE(exit, ok != 0)
+
+    uint8_t clear;
+    ok = fhe_uint8_decrypt(lhs, client_key, &clear);
+    GOTO_IF_TRUE(exit, ok != 0)
+
+    assert(clear == lhs_clear);
+    
+ exit:
+    fhe_uint8_destroy(lhs);
+    compressed_fhe_uint8_destroy(clhs);
+    fhe_uint8_destroy(result);
+  return ok;
+}
+
+int main(void)
```

Comment:
I'm going to be a bit annoying could we have a small encrypt/decrypt with u256 ?

Reply:
Added

Reply:
added one